src/test/java/io/strimzi/kafka/bridge/http/ProducerTest.java

```diff
@@ -644,4 +644,82 @@ void sendMultipleRecordsWithOneInvalidPartitionTest(VertxTestContext context) {
                     context.completeNow();
                 });
     }
+
+    @Test
+    void jsonPayloadTest(VertxTestContext context) {
+        String kafkaTopic = "breakOpenApiRules";
+        kafkaCluster.createTopic(kafkaTopic, 3, 1);
+
+        String value = "Hello from the other side";
+        String key = "message-key";
+
+        JsonArray records = new JsonArray();
+        JsonObject json = new JsonObject();
+        json.put("value", value);
+        json.put("key", key);
+        json.put("partition", 0);
+        records.add(json);
+
+        JsonObject root = new JsonObject();
+        root.put("records", records);
+
+        producerService()
+            .sendRecordsToPartitionRequest(kafkaTopic, 0, root, BridgeContentType.KAFKA_JSON_JSON)
+            .sendJsonObject(root, verifyBadRequest(context, "Validation error on: body.records[0] - " +
+                "$.records[0].partition: is not defined in the schema and the schema does not allow additional properties"));
+
+        records.remove(json);
```

Comment:
Just a nit: I would use `Json{Array|Object}.clear()` to be sure the object is empty. My memory is crappy and it is more complicated to follow what is in the object. However LGTM.

Reply:
Thanks for the comment and for the review :)